lib/mpl_toolkits/axisartist/tests/test_floating_axes.py

```diff
@@ -21,6 +21,9 @@ def test_subplot():
 # remove when image is regenerated.
 @image_comparison(['curvelinear3.png'], style='default', tol=5)
 def test_curvelinear3():
+    # Remove this lines when this test image is regenerated.
+    plt.rcParams.update({"xtick.direction": "in", "ytick.direction": "in"})
```

Comment:
If I get it correct, this changes the current behavior? Not sure if we should introduce some way to make that transition smoothly, but at least a change note would be in place, I guess.

Reply:
When we went to v2 we changed ticks from in to out, however because AxisArtist did not follow the rcparams they did not flip when everything else did.

I do not think there is a way to soften this change short of adding another rcparam.

I think we should go forward with this, but make sure there is a very clear API change note.

Reply:
I have added a deprecation warning at the level of Tick creation and relevant tests. On the other hand, users won't see the warning unless they manually create a Tick instance. I briefly thought about adding a `tickdir` keyword argument for Axes initialization, but not very inclined for now. I agree with @tacaswell that it is better to go with it.